File: src/main/java/project/rpg/items/potion/PotionBase.java
package project.rpg.items.potion;

import org.bukkit.event.player.PlayerItemConsumeEvent;
import project.rpg.Rpg;
import project.rpg.items.ItemBase;
import project.rpg.player.mana.Mana;

public abstract class PotionBase extends ItemBase {

    public abstract void onDrink(Mana mana, Rpg rpg, PlayerItemConsumeEvent event);

}

File: src/main/kotlin/project/rpg/items/potion/PotionBase.kt
package project.rpg.items.potion

import org.bukkit.event.player.PlayerItemConsumeEvent
import project.rpg.Rpg
import project.rpg.items.ItemBase
import project.rpg.player.mana.Mana

abstract class PotionBase : ItemBase() {
    abstract fun onDrink(mana: Mana, rpg: Rpg, event: PlayerItemConsumeEvent)
}